internal/Util.hpp

```diff
@@ -216,18 +216,25 @@ static bool loadMoFileStringsToArray(FILE* moFile,
 	return true;
 }
 
-static std::vector<std::string> buildMoFilePaths(const char *domain) {
+static std::vector<std::string> buildMoFilePaths(const char *domain)
+{
 	std::string languages = getenv("LANGUAGE");
 	std::vector<std::string> paths {""};
 	std::string slash = "/";
+	const size_t length = languages.size();
 	size_t pos = 0;
-	do {
-		pos = languages.find(":");
-		std::string lang = languages.substr(0, pos);
-		if (lang.empty()) break;
+	size_t colon = 0;
+	while (pos <= length)
```

Comment:
I think we can skip the last iteration with `while (pos != length)`. If `pos == length` the call to `find(':', pos)` will return `npos`, and `lang` will be empty.

Reply:
My concern here is, when `languages` only contains one locale (eg `en`), the only locale can be selected.
Also should we handle malformatted language strings, such as `:` `:en` `en:`, filter empty result before adding them?

Reply:
I like the way you have handled that in the last update. Too many colons result in empty strings, which are ignore. It is nice.

Reply:
... and since https://github.com/j-jorge/libintl-lite/pull/5#discussion_r769186515 , I think empty strings should be filtered out. If `moFilesRoot` is a directory, more empty strings would cause it to be opened and loaded more times.